dix/dispatch.c

```diff
@@ -2708,13 +2714,7 @@ ProcAllocColor(ClientPtr client)
         swapl(&rep.pixel);
     }
 
-#ifdef XINERAMA
-    if (noPanoramiXExtension || !pmap->pScreen->myNum)
```

Comment:
Shouldn’t we keep the logic of the `if (noPanoramiXExtension || !pmap->pScreen->myNum)` somehow?

Reply:
no, because ProcAllocColor() now is only called when no PanoramiX.
In the PanoramiX case, the reply is sent by PanoramiXAllocColor() (when it reached screen #0 - it's counting backwards).

but just did a little correction: not writing our secondary cmap's into the request buffer anymore, but using a stack variable instead.

Reply:
Do you know why the `!pmap->pScreen->myNum` check was needed?

Reply:
Yes, because the original PanoramiXAllocColor() implementation directly called into ProcAllocColor() unconditionally. That's exactly why I've factored out dixAllocColor() with this PR.